### ISSUE TITLE
new defect, striping

### ISSUE THREAD

**Issue by bobombolo** [bobombolo/preload#11]
![screenshot_20160407_210049](https://cloud.githubusercontent.com/assets/12679496/14373100/e8b1df98-fd03-11e5-92a6-6cdcdfe3adb4.png)

**Comment by Tank-ACiD** [comment]
yay pinstripes! it's like having a corduroy landscape?? ugg something is
just not lined up mathematically?

**Comment by bobombolo** [comment]
ya but nothing has changed besides the underlying Mag libs, my code wrt converting pixels to nodes is unchanged afaik...

**Comment by bobombolo** [comment]
good lighting though...

**Comment by bobombolo** [comment]
this is actually a shift of 15 of the 16 rows (z rows, rows in the raster) being shifted one block to the north. it is an effect of implementing Mags code for direct management of per-block set_node rather than map_interface level set_node. reverting back to the previous map_interface set_node goes back to the problem of holes #6

**Comment by bobombolo** [comment]
another clue is that the imported schems are in the correct place

![screenshot_20160407_233204](https://cloud.githubusercontent.com/assets/12679496/14374929/fffe7296-fd18-11e5-82e0-4797527ebd29.png)

![screenshot_20160407_233315](https://cloud.githubusercontent.com/assets/12679496/14374937/1e653eb8-fd19-11e5-8d6a-be3ebf7631fb.png)

**Comment by bobombolo** [comment]
hrm. is this an issue with the modulus in python returning the sign of the denominator?

https://github.com/bobombolo/preload/blob/master/preload.py#L194

http://stackoverflow.com/questions/10063546/modulo-for-negative-dividends-in-python

**Comment by bobombolo** [comment]
I have avoided the problem by calculating the in-block nodeID (the nomenclature is confusing), manually as is done in the LVM (only differently, lvm preferring zxy, here we use zyx) see latest diff. but I still think there is an error in the libminetest for calculating the nodeID with Pos.getAsInt() relating to negative numbers? this issue only manifested in our use case as a shift to the north because we are mapping an image which starts numbering at the top onto the southeast quadrant which is negative z, so we only ever have negative block or absolute node positions in the z portion... (none of our rasters have negative elevations yet)... at least this is my intuition.
manually calculating the nodeID is not any faster, since it involves arithmetic -- unlike the triple nested loop used with a simple counter in the lua vm object we are not actually looping through every node in the block (in the correct zyx order, or otherwise) but are instead building a block queue only as we encounter elevations.

however I also had to -1 on the block coordinates z value to get things to line up, which is odd. but this got the 16 node shift fixed. I don't think that was always there.
